pilopt/src/lib.rs

```diff
@@ -558,6 +563,129 @@ fn extract_constant_lookups<T: FieldElement>(pil_file: &mut Analyzed<T>) {
     }
 }
 
+/// Identifies witness columns that are constrained to a non-shifted (multi)linear expression, replaces the witness column by an intermediate polynomial.
```

Comment:
```suggestion
/// Identifies witness columns that are constrained to a non-shifted (multi)linear expression, replaces the witness column by an intermediate polynomial defined to be that expression.
```
(obvious, but this is what I was missing to understand it :) )

Reply:
Ah and maybe also use `intermediate column` instead of `polynomial`?